File: ClientApp/src/view-models/survey-result-view-model.ts
import { UserViewModel } from "./user-view-model";

export class SurveyResultViewModel {
  public code: string;
  public takenOn: string;
  public surveyId: number;
  public practitionerFullName: string;
  public practitionerId: string;
  public isCompleated: boolean;
  public firstStagePassed: boolean;
  public takersEmail: string;
  public surveyTakerUser: UserViewModel;

  constructor(code, takenOn, surveyId, practitionerFullName, practitionerId, isCompleated, takersEmail) {
    this.code = code;
    this.takenOn = takenOn;
    this.surveyId = surveyId;
    this.practitionerFullName = practitionerFullName;
    this.practitionerId = practitionerId;
    this.isCompleated = isCompleated;
    this.takersEmail = takersEmail;
  }

}

File: ViewModels/SurveyResultViewModel.cs
namespace KPProject.ViewModels
{
    public class SurveyResultViewModel
    {
        public string Code { get; set; }
        public string TakenOn { get; set; }
        public int SurveyId { get; set; }
        public string PractitionerFullName { get; set; }
        public string TakersEmail { get; set; }
        public string PractitionerId { get; set; }
        public bool IsCompleated { get; set; }
        public bool FirstStagePassed { get; set; }
        public UserViewModel SurveyTakerUser { get; set; }
    }
}
